Libraries/LibCrypto/PK/RSA.cpp

```diff
@@ -116,22 +123,46 @@ ErrorOr<RSA::KeyPairType> RSA::parse_rsa_key(ReadonlyBytes der, bool is_private,
 
 ErrorOr<RSA::KeyPairType> RSA::generate_key_pair(size_t bits, IntegerType e)
 {
-    IntegerType p;
-    IntegerType q;
-    IntegerType lambda;
+    auto ctx = TRY(OpenSSL_PKEY_CTX::wrap(EVP_PKEY_CTX_new_from_name(nullptr, "RSA", nullptr)));
```

Comment:
🤔 I don't quite understand what the difference between TRY(OpenSSL_FOO::wrap(operation)) and OPENSSL_TRY(operation) is supposed to be, is the first one just making a smart pointer?

Reply:
> is the first one just making a smart pointer?

Yes, that is something that was discussed in [my previous PR](https://github.com/LadybirdBrowser/ladybird/pull/3009): instead of using a bunch of scope guards to free memory we introduced some RAII wrappers. However for stuff that is lightly used, I haven't made wrappers (e.g. params builder).

Reply:
🤔 why not OwnPtr with a custom deleter then?
It also feels the wrong way around, I'd expect `OpenSSL_FOO::wrap(OPENSSL_TRY(operation))`?

Reply:
> why not OwnPtr with a custom deleter then?

We were unsure whether at some it might be useful to have methods on the wrappers to simply the most common OpenSSL patterns.

> It also feels the wrong way around

I've put it that way so that you cannot misuse it and not call `OPENSSL_TRY` because it is called internally in `::wrap`.

Reply:
Okay, thanks for clarifying!